### DISCUSSION TITLE
A bunch of question.

### DISCUSSION THREAD

**Discussion by Unlisted1723** [dani-garcia/vaultwarden#4080] [Q&A] [answered]
Hello there,

I just implement vaultwarden for my company. But some aspect are still not clear for me.

- If I end up with no network. How do my users access their password? Is there a cache where a key, who decrypts the password is stored? If that's so, how long does it exist? Is it best to use the browsers extension or the appliance?

- When someone creates a collection into an organisation. The person will have to add one by one each member of the organisation to get the right to read and modify the password. Is there a motorway to do so. Because if I have to give rights to all the member of my organisation, its gonna be long. Same if we have a new employee do we have to get him access to all the collection already in place? Is there an option to give maximum write to all the member of an organisation?

- In term of security. I was thinking of using a keepass. Who would be fed by vaultwarden every now and then. It would store in a disk plug into the network only when we export the data from vaultwarden. It would be of use if we loose internet or if for some reason we VM with vaultwarden is no more useable. Is it a good solution?

- If a user delete password by mistake. Can we get it back? Can we do a specific insert into the sqlite database. Or do we need to make a full recovery from our backup database?

There, it is all my question thanks for every bit of information you can give me.
Unlisted1723

**Comment by stefan0xC** [comment]
All of those questions should be answered by reading the [Bitwarden help pages](https://bitwarden.com/help/) as they are not really specific to Vaultwarden (which is only an alternative backend implementation).

Regarding using keepass: what would be your plan exactly?  :raised_eyebrow:

**Reply by devinslick** [reply, marked as answer]
Hey.  I'd suggest deploying a POC and thoroughly reading the documentation.  Not only will this address most of your questions, it may change your implementation.  
Regardless, I'll take a quick stab at answering your questions:

Network: clients sync/cache the credentials.  This cache, AFAIK, does not expire.

Browser extension:  depends on your use-case, but in most situations you'll need the extensions on a desktop.  Test them.

Provisioning users to Collections:  
Check out Groups.
https://bitwarden.com/help/about-groups/

Keypass:  No.  It sounds like you're trying to reinvent the client apps for offline use.  Test the bitwarden clients and browser extensions. 

Backup/restore:
Yes, you can restore individual items, depending on your configured policy.  This ability to restore shouldn't be confused with a backup though.   You should perform backups.
https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault

**Reply by Unlisted1723** [reply]
Thanks for those answers. I will look into it...

The idea of keepass born form the issues if we lose access to vaultwarden.
No longer access our vault and password.

Frome there, I don't know the name in English, but in French its a "sauvegarde morte" (literally meaning "dead backup").
So it would have been a SSD or a USB key bootable where keepass is installed. When every now and then we would export our vault bitwarden.
This SSD or USB key would then be stored into a physical save and use on an emergency purpose.

If you consider this unnecessary tell me, but it sound a good idea to me.
Thanks again for your answer,
Unlisted1723

**Reply by devinslick** [reply]
No problem, glad to help. If you lose access to the vaultwarden server your client apps will still have the credentials synced, so you'd still have the ability to use/export your data.
If you have a full server backup (as described in the link above), you can build a replacement vaultwarden server and the clients can continue syncing. 

Beyond that, if you want to keep your backups entirely offline, not connected/live (eg: dead), I'd suggest performing regular backups of the JSON/CSV data from your vault.  That file could be printed and put in a safe or stored on a drive that can be imported into any password manager.

Does that solve for the failure scenario(s) you're concerned about?
